### PR TITLE
Initial commit + lobby creation for sync start

### PR DESCRIPTION
**Pull request by teejusb**
Set up basic websockets using NestJS

Types are defined in `src/types/models.types.ts`
Main relevant code is define in `src/events/events.gateway.ts`
(client) testing is done in `src/events/events.gateway.spec.ts`

Currently we support the following:
- Player creating lobby
- Player joining lobby
- Player leaving a lobby
- Viewing existing lobbies

### CODE REVIEW

File: src/types/models.types.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+export class Judgments {
+  fantasticPlus: number;
+  fantastics: number;
+  excellents: number;
+  greats: number;
+  decents?: number;
+  wayOffs?: number;
+  misses: number;
```
Comment (Alhetus) [inline src/types/models.types.ts:8]: You'll probably want to add holds, rolls, mines etc. here as well

File: src/types/models.types.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  code: string;
+  // Empty string here is equivalent to "no password".
+  password: string;
+  players: Record<string, Player>;
+  spectators: Spectator[];
+}
+
+export class LobbyInfo {
+  code: string;
+  numberPlayers: number;
```
Comment (Alhetus) [inline src/types/models.types.ts]:
```suggestion
  playerCount: number;
```

File: src/types/models.types.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export class LOBBYMAN {
```
Comment (Alhetus) [inline src/types/models.types.ts:46]: oh god...

**Review by Alhetus — APPROVED** [review]
A few suggestions, overall looks like a very good starting point 👍

File: src/types/models.types.ts (hunk 3 above)
Comment (teejusb) [reply src/types/models.types.ts:46]: ty based stepmania

File: src/types/models.types.ts (hunk 2 above)
Comment (teejusb) [reply src/types/models.types.ts]: Done.

File: src/types/models.types.ts (hunk 1 above)
Comment (teejusb) [reply src/types/models.types.ts:8]: Done.

**Pull request merged**